### PR TITLE
Qingwei/hello world routing

### PR DESCRIPTION
**Pull request by qingweibinary**
Still not able to render the view ~
not sure if I understand the routing correctly

### CODE REVIEW

File: src/_reducers/HelloWorldReducer.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import StateStorage from '../_store/StateStorage';
```
Comment (borisyankov) [inline src/_reducers/HelloWorldReducer.js]: This is a custom built thing by me, we should be replacing it with redux-localstorage (which seems to be better).

File: src/_reducers/HelloWorldReducer.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const SayHello = 'Hello';
+const SayHi = 'Hi';
+const NoSayHello = '';
+
+const initialState = new Map({subject: 'World'});
```
Comment (borisyankov) [inline src/_reducers/HelloWorldReducer.js]: You need to import 'immutable' or else you are using the ES6 Map. Which will lead to unexpected behavior ;) One solution is to always use Immutable.Map, another is to continue being cool and using it like this but being mindful. ;)

File: src/_reducers/HelloWorldReducer.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+export default (state=initialState, action=null) => {
```
Comment (borisyankov) [inline src/_reducers/HelloWorldReducer.js]: I don't think action = null is necessary, the default will be undefinied which is almost equivalent.
The rest looks correct.

File: src/_reducers/HelloWorldReducer.js (hunk 1 above)
Comment (qingweibinary) [inline src/_reducers/HelloWorldReducer.js]: ok, now I get what you meant previously

File: src/_routes/index.js [hunk 4]
```diff
@@ -4,7 +4,9 @@ import TickTradeRoutes from './TickTradeRoutes';
 import DesktopRoutes from './DesktopRoutes';
 import CommonRoutes from './CommonRoutes';
 
-const childRoutes = [].concat(DesktopRoutes).concat(TickTradeRoutes).concat(CommonRoutes);
+import HelloWorldRoute from './HelloWorldRoute'
+
+const childRoutes = [].concat(DesktopRoutes).concat(TickTradeRoutes).concat(CommonRoutes).concat(HelloWorldRoute);
```
Comment (borisyankov) [inline src/_routes/index.js]: You can't concat an object to an array, you would have to do .concat([HelloWorldRoute])

File: src/helloworld/HelloWorldPanel.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        subject: String
+    }
+
+    render() {
+        return
+        <div>
+            <p>{this.props.greet + " " + this.props.subject}</p>
+        </div>
+    }
+}
```
Comment (borisyankov) [inline src/helloworld/HelloWorldPanel.js]: Looks correct

File: src/_reducers/HelloWorldReducer.js (hunk 2 above)
Comment (qingweibinary) [inline src/_reducers/HelloWorldReducer.js]: yea, thanks for catching

**Pull request closed without merging**